### ISSUE TITLE
Support R factor data type

### ISSUE THREAD

**Issue by BoPeng** [vatlab/sos-r#18]
http://pandas-docs.github.io/pandas-docs-travis/user_guide/categorical.html

Pandas now has categorical data type.

**Comment by rezaeir** [comment]
I just use R built-in `InsectSprays` dataset which has a factor column and used `%get` to import it into SoS and it imported the dataframe without problem and the factor column is interpreted as `category` in pandas. Moreover, I deleted this dataset from my R variables and imported it from SoS and it recognized the column as `factor`. It doesn't seem to have any problem. What should be fixed here?

**Comment by BoPeng** [comment]
Your tests work because dataframes are passed through `feather/arrow`. You should notice problem if you create a categorical variable and pass it directly.

**Comment by rezaeir** [comment]
I just checked that!! it doesn't import the variable and just assigns the variable name to `'Untransferrable variable'` string. It seems like there is a specific function handling this, because I'm not familiar with SoS code, could you please give me a link to that specific function or should I find it myself?

**Comment by BoPeng** [comment]
You need to first read https://vatlab.github.io/sos-docs/doc/user_guide/language_module.html and then dive into [the source code of sos-r](https://github.com/vatlab/sos-r/blob/master/src/sos_r/kernel.py). I think support for factor on both ends are missing.

**Comment by rezaeir** [comment]
usually, R's vector should translate to python's bulit-in list. However, as much as python lists are not able to handle categorical data (correct me if I'm wrong and I search for a way to transfer data to list). My current idea is to translate R's factor vector to a pandas series and if someone writes a categorical pandas series, I translate it into a factor R vector. What do you think?

**Comment by BoPeng** [comment]
I agree, R's category type matches perfectly to Panda's category series http://pandas-docs.github.io/pandas-docs-travis/user_guide/categorical.html , and vise versa.

**Comment by rezaeir** [comment]
I tried to test all the possible factors or pandas categories in my jupyterlab but I think there should be some predetermined tests for it!!
